### DISCUSSION TITLE
Research git notes

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#22213] [Suggest an Idea]
### Describe the proposed change(s).

Possibly we could use git notes for "state" storage.

Refs:
- https://git-scm.com/docs/git-notes
- https://stackoverflow.com/questions/2683248/can-i-add-metadata-to-git-commits-or-can-i-hide-some-tags-in-gitk

Related:
- https://stackoverflow.com/questions/54805875/git-push-all-objects-to-remote-repository-not-just-refs-but-absolutely-all-bl
- https://www.kabisa.nl/tech/git-content-based-addressing-branches-and-tags/

**Comment by viceice** [comment]
Will this work with upcoming GitHub API commits?

**Comment by rarkins** [comment]
Probably not :)

**Comment by viceice** [comment]
> Probably not :)

Maybe we can fetch the API commit and add and push the notes after that. 🤔
